src/goto-instrument/contracts/havoc_assigns_clause_targets.cpp

```diff
@@ -211,38 +265,45 @@ void havoc_assigns_clause_targets(
 
   for(const auto &target : targets)
   {
-    const auto &tracking_comment =
-      make_tracking_comment(target, replaced_function_id, ns);
-
-    const auto target_pointer = build_address_of(target, ns);
-    const auto target_snapshot_var = new_tmp_symbol(
-                                       target_pointer.type(),
-                                       source_location,
-                                       mode,
-                                       st,
-                                       "__target_snapshot_var",
-                                       true)
-                                       .symbol_expr();
-    const auto target_valid_var =
-      new_tmp_symbol(
-        bool_typet(), source_location, mode, st, "__target_valid_var", true)
-        .symbol_expr();
-
-    snapshot_if_valid(
-      target_valid_var,
-      target_snapshot_var,
-      target_pointer,
-      snapshot_program,
-      source_location,
-      ns);
-    havoc_if_valid(
-      target_valid_var,
-      target_snapshot_var,
-      target,
-      tracking_comment,
-      havoc_program,
-      source_location,
-      ns);
+    if(can_cast_expr<conditional_target_exprt>(target))
+    {
+      const auto &cond_target = to_conditional_target_expr(target);
+
+      // clean the condition if needed
+      null_message_handlert null_handler;
```

Comment:
Here and in the other place that introduces a `null_message_handlert`: should you perhaps use `log.get_message_handler()` from code_contractst?

Reply:
fixed